core/src/main/java/hudson/model/Renamable.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2023 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package hudson.model;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.util.FormValidation;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.ProtectedExternally;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.QueryParameter;
+
+/**
+ * Interface used to create a dedicated page that changes the name property of an Object.
+ * See also {@link jenkins.model.RenameAction}.
+ *
+ * @since TODO
+ */
+@Restricted(ProtectedExternally.class)
+public interface Renamable {
```

Comment:
Having a hard time understanding the relationship between this interface and `Node`/`Slave`. Should they implement this interface or not? If so, why don't they? If not, why does renaming work without issue there?

Reply:
The idea was to make them implement this interface as well: https://github.com/jenkinsci/jenkins/pull/8310#issuecomment-1682392118

I had originally tried to do it the same way as [Computer](https://github.com/jenkinsci/jenkins/blob/c5fbbbe88dc280d0a0876614c0f9aa6fba2ea2af/core/src/main/java/hudson/model/Computer.java#L1517), but it always failed testing it in other plugins such as `ec2` and `azure-vm-agents`.
Looking back now, I realize the problem was always the `cloudName` vs `name` issue that was finally identified and corrected. I have just pushed up a second, simpler, version of a fix for this bug. See #8505

Reply:
To fully answer the original comment, I guess it is more a matter of being consistent across our UIs? Changing the name of a `Node` isn't an issue because there is no `Apply` button. Should we implement this interface in `Node`, the apply button could be added back in.
Given how this PR affected a bunch of downstream PRs, it's probably best to implement this interface one component at a time.

Reply:
On the flip side, why not remove the **Apply** button from `Cloud` to make it consistent with `Node`? That's inconsistent with `AbstractItem`, but so is the status quo where `Node` is inconsistent with `AbstractItem`. And at least it fixes the bug.

A request to defer implementing this interface in `Cloud` is effectively a request to introduce debt. I would not automatically reject such a request, but I think it needs to be justified, and merely mentioning "how this PR affected a bunch of downstream PRs" does not seem like a valid justification to me. And I think the intention behind the request is to implement in stages (which is fine, _if_ the rest of the stages are actually finished in a timely fashion, something which almost never happens in my experience whenever someone makes such a request), but an unintentional consequence is that the design of the interface might turn out worse.

When defining a new interface, it is important to consume it in at least three places before shipping it to ensure the design is correct. If you implement it only once, it probably won't support a second implementation. If you implement it only twice, it will probably support a third implementation with difficulty. If you implement it three or more times, the interface will probably work fine. Will Tracz calls this "The Rule of Threes" (_Confessions of a Used Program Salesman_, Addison-Wesley, 1995).

So my perspective is that the third implementation should be in scope for this PR not only in order to make the implementation complete but also to validate the design of the interface.